addons/account/models/account_move.py

```diff
@@ -1437,6 +1437,30 @@ def _get_invoice_in_payment_state(self):
         'line_ids.payment_id.state',
         'line_ids.full_reconcile_id')
     def _compute_amount(self):
+        in_invoices = self.filtered(lambda m: m.move_type == 'in_invoice')
+        out_invoices = self.filtered(lambda m: m.move_type == 'out_invoice')
+        others = self.filtered(lambda m: m.move_type not in ('in_invoice', 'out_invoice'))
+        reversed_mapping = defaultdict(lambda: self.env['account.move'])
+        for reverse_move in self.env['account.move'].search([
+            ('state', '=', 'posted'),
+            '|', '|',
+            '&', ('reversed_entry_id', 'in', in_invoices.ids), ('move_type', '=', 'in_refund'),
+            '&', ('reversed_entry_id', 'in', out_invoices.ids), ('move_type', '=', 'out_refund'),
+            '&', ('reversed_entry_id', 'in', others.ids), ('move_type', '=', 'entry'),
+        ]):
+            reversed_mapping[reverse_move.reversed_entry_id] += reverse_move
+
+        caba_mapping = defaultdict(lambda: self.env['account.move'])
+        caba_company_ids = self.company_id.filtered(lambda c: c.tax_exigibility)
+        reverse_moves_ids = [move.id for moves in reversed_mapping.values() for move in moves]
+        for caba_move in self.env['account.move'].search([
+            ('tax_cash_basis_origin_move_id', 'in', self.ids + reverse_moves_ids),
+            ('state', '=', 'posted'),
+            ('move_type', '=', 'entry'),
+            ('company_id', 'in', caba_company_ids.ids)
+        ]):
+            caba_mapping[caba_move.tax_cash_basis_origin_move_id] += caba_move
```

Comment:
Regarding the "performance" test that failed, you could do this search only if `caba_company_ids` is not falsy

For the first search, it is of course "possible" to do it only when the state is going to be `paid`, but I thing that it is overkill